Test unwrapped DBus values

poncovka opened the pull request:
The DBus library only unwraps the received DBus values now, so we don't have
to use the unpacked form of DBus values in the unit tests anymore.

The DBus library raises DBusError from dasbus.error by default now. All Anaconda
DBus errors should be descendants of DBusError, so they are easier to handle.

**Depends on:** https://github.com/rhinstaller/dasbus/pull/13

jkonecny12 commented on tests/nosetests/pyanaconda_tests/module_part_manual_test.py:
```diff
@@ -78,93 +78,54 @@ def mount_points_property_test(self):
+                    "format-options": get_variant(Str, ""),
+                    "mount-options": get_variant(Str, "")
+                }
+            ]
```
These and similar changes are not that self explanatory as the original variant. When you have `in_value` and `out_value` it's immediately clear what that means but adding `[ { ... } ]` does not say anything. Could you please add naming or comment?

poncovka replied:
Fixed.

jkonecny12 commented on tests/nosetests/pyanaconda_tests/module_security_test.py:
```diff
@@ -50,6 +50,14 @@ def setUp(self):
         self.callback = PropertiesChangedCallback()
         self.security_interface.PropertiesChanged.connect(self.callback)
 
+    def _test_dbus_property(self, *args, **kwargs):
```
Please use the name `_check_dbus_property()`. The `test` prefix is used for test cases so it's better to use check for helper logic.

poncovka replied:
Fixed.

jkonecny12 commented on tests/nosetests/pyanaconda_tests/__init__.py:
```diff
@@ -189,29 +188,36 @@ def check_task_creation_list(test, task_paths, publisher, task_classes):
     """
-    publisher.assert_called_once()
-    object_path, obj = publisher.call_args[0]
+    test.assertGreater(publisher.call_count, index)
```
Could you please add explanation why the call count have to be greater than index? Is that because index starts from 0 but call count is counted from 1?

poncovka replied:
Fixed.

jkonecny12 commented on tests/nosetests/pyanaconda_tests/__init__.py:
```diff
@@ -189,29 +188,36 @@ def check_task_creation_list(test, task_paths, publisher, task_classes):
+    :param index: an index of the published object
     """
-    publisher.assert_called_once()
-    object_path, obj = publisher.call_args[0]
```
Do I understand it correctly that you are fixing the `check_task_creation_list`? If yes than please write that in commit message.

poncovka replied:
Fixed.

jkonecny12 commented:
jenkins, test this please

poncovka commented:
Rebased, fixed a small bug with https://github.com/rhinstaller/anaconda/pull/2256/commits/4de7c3a7b1dbdd05736073fb63a10d4520879968 and changed the required version of dasbus.

VladimirSlavik commented on tests/nosetests/pyanaconda_tests/module_security_test.py:
```diff
@@ -173,30 +175,20 @@ def realm_kickstart_test(self):
         """Test module in default state with realm discover task."""
         realm_discover_task_path = self.security_interface.DiscoverRealmWithTask()
-
-        publisher.assert_called()
```
Just wondering aloud: All the `assert_called` were redundant?

poncovka replied:
Yes, they are. The publisher is checked in `check_task_creation`.
